src/inmanta/db/versions/v202509100.py

```diff
@@ -33,6 +33,15 @@ async def update(connection: Connection) -> None:
         AND rscm.resource_set_id=rs.id
     );
 
+    -- Remove rps entries that no longer have a matching resource in the database
+    DELETE FROM public.resource_persistent_state AS rps
+    WHERE NOT EXISTS (
+        SELECT 1
+        FROM resource AS r
+        WHERE rps.environment=r.environment
+        AND rps.resource_id=r.resource_id
+    );
+
```

Comment:
We need to verify the performance of this query. I will already approve it, so that you can merge it in when it's verified.